lib/Doctrine/DBAL/Driver/ResultStatement.php

```diff
@@ -109,7 +109,7 @@ public function fetchAll($fetchMode = null, $fetchArgument = null, $ctorArgs = n
      *                         If no value is supplied, PDOStatement->fetchColumn()
      *                         fetches the first column.
      *
-     * @return string|boolean A single column in the next row of a result set, or FALSE if there are no more rows.
+     * @return string|boolean|null A single column in the next row of a result set, or FALSE if there are no more rows.
```

Comment:
When can `null` be returned? Can we have a scenario where this is tested?

Reply:
When you have an aggregate query (like `SELECT AVG`) and it does not match any rows. It might be MySQL-specific.

I'm not familiar with DBAL tests so not sure how to add it.

Reply:
Probably somewhere like in https://github.com/doctrine/dbal/blob/66daecd0084118ec04c54288915a8eb9bac7511d/tests/Doctrine/Tests/DBAL/Functional/PDOStatementTest.php (or the `Functional` dir - didn't pin-point it to a specific test)

Reply:
OK, I'll look into it 😊

Reply:
Here's a simple example:
```php
$value = $conn->executeQuery(
    $conn->getDatabasePlatform()
        ->getDummySelectSQL('NULL')
)->fetchColumn();

var_dump($value);
// NULL
```